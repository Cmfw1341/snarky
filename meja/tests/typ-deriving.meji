type t = {a: bool, b: field};

type u('a) = {a1: 'a, b1: bool};

convertible type v('a, 'b) = {a2: 'a, b2: 'b}
with prover v('a, 'b) = {a2: 'a, b2: 'b}
by v_typ;

let x : _ -> v(_, _);

let y : unit -> v(boolean, boolean);


