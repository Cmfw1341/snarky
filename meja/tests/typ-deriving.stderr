File "tests/typ-deriving.meja", line 1, characters 0-30:
Error: Unbound type constructor Field.Var.t.
